ipykernel/kernelbase.py

```diff
@@ -528,6 +492,19 @@ async def process_one(self, wait=True):
                 t, dispatch, args = self.msg_queue.get_nowait()
             except (asyncio.QueueEmpty, QueueEmpty):
                 return
+
+        if self.control_thread is None and self.control_stream is not None:
+            # If there isn't a separate control thread then this main thread handles both shell
+            # and control messages. Before processing a shell message need to flush all control
+            # messages and allow them all to be processed.
+            await asyncio.sleep(0)
+            self.control_stream.flush()
+
+            socket = self.control_stream.socket
+            while socket.poll(1):
```

Comment:
This is blocking for 1ms, right? I think it should just check if there is a message:
```suggestion
            while socket.poll(0):
```

Reply:
I'd like to, but if I do that the `ipyparallel` tests fail locally for me.